tracer/src/Datadog.Trace/DuckTyping/DuckTypeExtensions.cs

```diff
@@ -204,19 +171,12 @@ public static object DuckImplement(this object instance, Type typeToDeriveFrom)
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static bool TryDuckImplement(this object? instance, Type? typeToDeriveFrom, [NotNullWhen(true)] out object? value)
         {
-            if (instance is null)
+            if (instance is not null &&
+                typeToDeriveFrom is not null &&
```

Comment:
When would `typeToDeriveFrom` be `null`? Why is that parameter nullable? 🤔

Reply:
I did some changes in https://github.com/DataDog/dd-trace-dotnet/pull/4582/commits/fbb089537a2198a51d6a7faf802de611993dc3c6